index.d.ts

```diff
@@ -13,32 +13,104 @@ export function tracingSubscriber(): void
+   * Metadata is made available to you in the ngrok dashboard and the Agents API
+   * resource. It is a useful way to allow you to uniquely identify sessions. We
+   * suggest encoding the value in a structured format like JSON.
+   *
+   * See the [metdata parameter in the ngrok docs] for additional details.
+   *
+   * [metdata parameter in the ngrok docs]: https://ngrok.com/docs/ngrok-agent/config#metadata
    */
   metadata(metadata: string): this
-  /** Connect to the provided ngrok server address. */
+  /**
+   * Configures the network address to dial to connect to the ngrok service.
+   * Use this option only if you are connecting to a custom agent ingress.
+   *
+   * See the [server_addr parameter in the ngrok docs] for additional details.
+   *
+   * [server_addr parameter in the ngrok docs]: https://ngrok.com/docs/ngrok-agent/config#server_addr
+   */
   serverAddr(addr: string): this
+  /**
+   * Configures a function which is called when the ngrok service requests that
+   * this [Session] stops. Your application may choose to interpret this callback
+   * as a request to terminate the [Session] or the entire process.
+   *
+   * Errors returned by this function will be visible to the ngrok dashboard or
+   * API as the response to the Stop operation.
+   *
+   * Do not block inside this callback. It will cause the Dashboard or API
+   * stop operation to time out. Do not call [std::process::exit] inside this
+   * callback, it will also cause the operation to time out.
+   */
+  handleStopCommand(handler: () => void): this
+  /**
+   * Configures a function which is called when the ngrok service requests
+   * that this [Session] updates. Your application may choose to interpret
+   * this callback as a request to restart the [Session] or the entire
+   * process.
+   *
+   * Errors returned by this function will be visible to the ngrok dashboard or
+   * API as the response to the Restart operation.
+   *
+   * Do not block inside this callback. It will cause the Dashboard or API
+   * stop operation to time out. Do not call [std::process::exit] inside this
+   * callback, it will also cause the operation to time out.
+   */
+  handleRestartCommand(handler: () => void): this
+  /**
+   * Configures a function which is called when the ngrok service requests
+   * that this [Session] updates. Your application may choose to interpret
+   * this callback as a request to update its configuration, itself, or to
+   * invoke some other application-specific behavior.
+   *
+   * Errors returned by this function will be visible to the ngrok dashboard or
+   * API as the response to the Restart operation.
+   *
+   * Do not block inside this callback. It will cause the Dashboard or API
+   * stop operation to time out. Do not call [std::process::exit] inside this
+   * callback, it will also cause the operation to time out.
+   */
+  handleUpdateCommand(handler: (err: null|Error, update: UpdateRequest) => void): this
```

Comment:
This signature seems weird. Where is that `err` argument coming from?

Reply:
It's injected by napi into the callback function invocation, first argument is always an optional error, since it's called with a Result. I can research if there's a way to avoid that since in this case we're always a-ok.

Reply:
ah, `ErrorStrategy::Fatal` is probably the ticket, will modify.

Reply:
Pushed update, makes it cleaner, thanks for bringing up!